tools/jdbc/src/jni/duckdb_java.cpp

```diff
@@ -785,6 +785,7 @@ JNIEXPORT jobjectArray JNICALL Java_org_duckdb_DuckDBNative_duckdb_1jdbc_1fetch(
 		case LogicalTypeId::DOUBLE:
 			constlen_data = env->NewDirectByteBuffer(FlatVector::GetData(vec), row_count * sizeof(double));
 			break;
+		case LogicalTypeId::TIME_TZ:
 		case LogicalTypeId::TIMESTAMP_SEC:
 		case LogicalTypeId::TIMESTAMP_MS:
```

Comment:
Unrelated to this PR but I wonder if the non-microsecond timestamp types are handled correctly. It seems like they use the exact same code path as the regular timestamps - which will likely result in incorrect conversions. The types are almost never used outside of conversions from specific formats (e.g. Arrow) so it is not that unthinkable that these have not been correctly tested in the JDBC client.

Perhaps it makes sense to add the `test_all_types` test to the JDBC client as well?